website/common/script/ops/scoreTask.js

```diff
@@ -333,13 +341,12 @@ export default function scoreTask (options = {}, req = {}) {
   } else if (task.type === 'reward') {
     // Don't adjust values for rewards
     delta += _changeTaskValue(user, task, direction, times, cron);
+    // Since historically some users can have rewards with negative cost (which will cause
+    // validation failure), in order to 'buy' a reward and update it in database we need to ensure
+    // the reward cost is a positive number
+    task.value = Math.abs(task.value);
```

Comment:
what do you think of this @Alys ? Basically if you have a reward with a negative value it'll be treated as if the value is positive, so a value of -5 will be treated as +5